subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/RealLifeAndroidDexingTransformsPerformanceTest.groovy

```diff
@@ -0,0 +1,157 @@
+            }
+        }
+
+        when:
+        def results = runner.run()
+
+        then:
+        results.assertEveryBuildSucceeds()
+        and:
+        assertDexingTransformIsFaster(results)
+
+        where:
+        testProject         | memory | warmUpRuns | runs | tasks
+        'largeAndroidBuild' | '5g'   | 2          | 4    | 'clean assembleDebug'
+    }
+
+    @Unroll
+    def "dexing task vs transform: #tasks on #testProject"() {
+        given:
+        def dexingTransform = DEXING_TRANSFORM
+        def dexingTask = DEXING_TASK
+
+        def invocationOptions = [tasks: tasks, memory: memory, enableAndroidBuildCache: true]
+
+        runner.testGroup = "Android dexing"
+        runner.buildSpec {
+            projectName(testProject)
+            displayName(dexingTransform)
+            warmUpCount warmUpRuns
+            invocationCount runs
+            invocation {
+                defaultInvocation(*:invocationOptions, dexingTransforms: true, delegate)
+            }
+        }
+
+        runner.baseline {
+            projectName(testProject)
+            displayName(dexingTask)
+            warmUpCount warmUpRuns
+            invocationCount runs
+            invocation {
+                defaultInvocation(*:invocationOptions, dexingTransforms: false, delegate)
+            }
+        }
+
+        when:
+        def results = runner.run()
+
+        then:
+        results.assertEveryBuildSucceeds()
+        and:
+        assertDexingTransformIsFaster(results)
+
+        where:
+        testProject         | memory | warmUpRuns | runs | tasks
+        'largeAndroidBuild' | '5g'   | 2          | 4    | 'clean assembleDebug'
+    }
+
+    private static BuildExperimentListenerAdapter cleanTransformsCache() {
+        new BuildExperimentListenerAdapter() {
+            @Override
+            void beforeInvocation(BuildExperimentInvocationInfo invocationInfo) {
+                GFileUtils.deleteDirectory(new File(invocationInfo.gradleUserHome, "caches/transforms-1/files-1.1"))
+            }
+        }
+    }
+
+    void defaultInvocation(Map options, GradleInvocationSpec.InvocationBuilder builder) {
+        String memory = options.memory
+        String[] tasks = options.tasks.toString().split(' ')
+        with(builder) {
+            tasksToRun(tasks)
+            cleanTasks("clean")
+            gradleOpts("-Xms${memory}", "-Xmx${memory}")
+            useDaemon()
+            args("-Dorg.gradle.parallel=true", "-Pandroid.enableBuildCache=${options.enableAndroidBuildCache ?: true}", "-Pandroid.enableDexingArtifactTransform=${options.dexingTransforms}", '-Dcom.android.build.gradle.overrideVersionCheck=true')
+        }
+
+    }
+
+    private static void assertDexingTransformIsFaster(CrossBuildPerformanceResults results) {
+        def transformResults = results.buildResult(DEXING_TRANSFORM)
+        BaselineVersion taskResults = dexingTaskResult(results)
```

Comment:
It would be nice to have these two lines look similar as they seem to be doing similar things. Extracting `results.buildResult(DEXING_TRANSFORM)` into its own `dexingTransformResult()` method would be nice.

Maybe renaming the methods to `getDexingByTaskResults()` and `getDexingByTransformResults()` could clarify things a bit, too.

Reply:
They are more similar now.